pkg/controllers/allocation/controller.go

```diff
@@ -105,5 +127,50 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 			zap.S().Errorf("Continuing after failing to bind, %s", err.Error())
 		}
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
+	return reconcile.Result{}, nil
+}
+
+// Watches returns the necessary information to create a watch
+//   a. source: the resource that is being watched
+//   b. eventHandler: which controller objects to be reconciled
+//   c. predicates: which events can be filtered out before processed
+func (c *Controller) Watches(ctx context.Context) (source.Source, handler.EventHandler, builder.WatchesOption) {
+	return &source.Kind{Type: &v1.Pod{}},
+		handler.EnqueueRequestsFromMapFunc(handler.MapFunc(
+			func(obj client.Object) []reconcile.Request {
+				pod := obj.(*v1.Pod)
+				if pod.Spec.NodeName != "" || podutils.IsOwnedByDaemonSet(pod) {
```

Comment:
Can you share these checks if the filter function somehow?

Reply:
actually, I think I can delete this all together since `isProvisionable` is already checked in the UpdateFunc predicate and that should cover the nodeName being nil since it checks if the pod failed to schedule.

Reply:
Thought on inverting this decision and putting it all in the map func, rather than the predicate func? 

I think controller-runtime's programming model is way too complex here.

Reply:
yeah, that's probably best, I'll move it